HerPortal/Controllers/CsvFileController.cs

```diff
@@ -13,36 +12,29 @@ namespace HerPortal.Controllers;
 [Route("/download")]
 public class CsvFileController : Controller
 {
-    private readonly UserDataStore userDataStore;
-    private readonly ICsvFileGetter csvFileGetter;
+    private readonly ICsvFileService csvFileService;
     private readonly ILogger<CsvFileController> logger;
 
     public CsvFileController
     (
-        UserDataStore userDataStore,
-        ICsvFileGetter csvFileGetter,
+        ICsvFileService csvFileService,
         ILogger<CsvFileController> logger
     ) {
-        this.userDataStore = userDataStore;
-        this.csvFileGetter = csvFileGetter;
+        this.csvFileService = csvFileService;
         this.logger = logger;
     }
     
     [HttpGet("{custodianCode}/{year:int}/{month:int}")]
     public async Task<IActionResult> GetCsvFile(string custodianCode, int year, int month)
     {
-        // Important! First ensure the logged-in user is allowed to access this data
-        var userEmailAddress = HttpContext.User.GetEmailAddress();
-        var userData = await userDataStore.GetUserByEmailAsync(userEmailAddress);
-        if (!userData.LocalAuthorities.Any(la => la.CustodianCode == custodianCode))
-        {
-            return Unauthorized("The logged-in user is not permitted to access this resource.");
-        }
-
         Stream file;
         try
         {
-            file = await csvFileGetter.GetFileForDownloadAsync(custodianCode, year, month, userData.Id);
+            file = await csvFileService.GetFileForDownloadAsync(custodianCode, year, month, HttpContext.User.GetEmailAddress());
+        }
+        catch (SecurityException)
+        {
+            return Unauthorized("The logged-in user is not permitted to access this resource.");
```

Comment:
Note that this loses all details of the original exception - e.g. the message and stack trace. It's not too bad because we only throw a `SecurityException` in one place, but it might make it harder to debug for someone unfamiliar with it?